cmake/modules/AddCephTest.cmake

```diff
@@ -70,14 +70,12 @@ function(add_tox_test name)
   endif()
   string(REPLACE ";" "," tox_envs "${tox_envs}")
   find_package(Python3 QUIET REQUIRED)
-  add_custom_command(
-    OUTPUT ${venv_path}/bin/activate
-    COMMAND ${CMAKE_SOURCE_DIR}/src/tools/setup-virtualenv.sh --python="${Python3_EXECUTABLE}" ${venv_path}
-    WORKING_DIRECTORY ${tox_path}
-    COMMENT "preparing venv for ${name}")
-  add_custom_target(${name}-venv
-    DEPENDS ${venv_path}/bin/activate)
-  add_dependencies(tests ${name}-venv)
+  add_test(
+    NAME setup-venv-for-${name}
+    COMMAND ${CMAKE_SOURCE_DIR}/src/tools/setup-virtualenv.sh --python=${Python3_EXECUTABLE} ${venv_path}
+    WORKING_DIRECTORY ${tox_path})
+  set_tests_properties(setup-venv-for-${name} PROPERTIES
+    FIXTURES_SETUP venv-for-${name})
```

Comment:
Nice thing @tchaikov. Shan't we need to add a [FIXTURE_CLEANUP](FIXTURES_CLEANUP) for taking care of venvs?

Reply:
i thought about adding a  [FIXTURES_CLEANUP](https://cmake.org/cmake/help/latest/prop_test/FIXTURES_CLEANUP.html) step. but i was also thinking about the use case where developers need to rerun the tests repeatedly. if you think it's fine to teardown and recreate the venv in that case, i will add a FIXTURES_CLEANUP to nuke the venv. what do you think?

Reply:
If `ninja clean` (or whatever) takes care of those, I think we can leave them created for quick retesting.

Reply:
@epuertat `clean` removes the files created by ninja rules. but `add_test()` does not allow me to specify any side product. so i added a `FIXTURES_CLEANUP` for the venv.